src/ripple/rpc/impl/TransactionSign.cpp

```diff
@@ -335,18 +336,19 @@ transactionSign (
 
     WriteLog (lsDEBUG, RPCHandler) << "transactionSign: " << params;
 
-    if (! params.isMember ("secret"))
-        return RPC::missing_field_error ("secret");
+    SignatureKeypair keypair;
```

Comment:
Ah, exceptions are so sweet... :-)  I personally think that this is much more logical a flow control.